CONTRIBUTING.md

```diff
@@ -4,6 +4,24 @@ This document should concisely express the project development status,
 methodology, and contribution process.  As the community makes progress, we
 should keep this document in sync with reality.
 
+## Submitting a Pull Request (PR)
+
+The following outlines the general rules we follow:
+
+- All PRs must have the appropriate documentation changes made within the
+same PR. Note, not all PRs will necessarily require a documentation change
+but if it does please include it in the same PR so the PR is complete and
+not just a partial solution.
+- All PRs must have the appropriate testcases. For example, bug fixes should
```

Comment:
All of the "should's" and "as possible" were meant to provide the "out" you're looking for - obviously there's wiggle room that reviewers have. The point was to encourage (w/o mandating) a high bar. However, I use your preferred "with reason" text since you seem to prefer that.